test/integration/build_context_dir_test.go

```diff
@@ -0,0 +1,75 @@
+// +build integration,extended,!no-etcd,!no-docker,docker
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+)
+
+func init() {
+	requireDocker()
+	requireEtcd()
+}
+
+func TestExtendedBuildContextDir(t *testing.T) {
+	deleteAllEtcdKeys()
+	openshift, _ := StartTestAllInOne()
+	client, _, err := openshift.GetOpenshiftClient()
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	build := mockBuildWithContextDir()
+	if _, err := client.Builds(testNamespace).Create(build); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	retries := 0
+
+	for {
+		time.Sleep(time.Duration(1 * time.Second))
```

Comment:
FYI. this should be probably solved by using Watch() instead of polling, I just need to get familiar with the Watch() itself first...